src/_pytest/main.py

```diff
@@ -877,22 +877,24 @@ def resolve_collection_argument(
     If the path doesn't exist, raise UsageError.
     If the path is a directory and selection parts are present, raise UsageError.
     """
-    base, squacket, rest = str(arg).partition("[")
-    strpath, *parts = base.split("::")
-    if parts:
+    strpath, selection, rest = arg.partition("::")
+    test_name, squacket, rest = rest.partition("[")
+    parts = []
+
+    if test_name:
+        parts = test_name.split("::")
         parts[-1] = f"{parts[-1]}{squacket}{rest}"
     if as_pypath:
         strpath = search_pypath(strpath)
-    fspath = invocation_path / strpath
-    fspath = absolutepath(fspath)
+    fspath = absolutepath(invocation_path / strpath)
```

Comment:
Again in the interest of keeping the change minimal (and keeping the "git blame" in tact), would be better to drop this cosmetic change.